src/control/lib/netdetect/netdetect.go

```diff
@@ -887,45 +1013,64 @@ func ScanFabric(provider string) ([]FabricScan, error) {
 	log.Debugf("initDeviceScan completed.  Depth %d, numObj %d, systemDeviceNames %v, hwlocDeviceNames %v",
 		deviceScanCfg.depth, deviceScanCfg.numObj, deviceScanCfg.systemDeviceNames, deviceScanCfg.hwlocDeviceNames)
 
+	hfiDeviceCount := getHFIDeviceCount(deviceScanCfg.hwlocDeviceNames)
+	log.Debugf("There are %d hfi1 devices in the system", hfiDeviceCount)
+
 	for ; fi != nil; fi = fi.next {
 		if fi.domain_attr == nil || fi.domain_attr.name == nil || fi.fabric_attr == nil || fi.fabric_attr.prov_name == nil {
 			continue
 		}
-
 		deviceScanCfg.targetDevice = C.GoString(fi.domain_attr.name)
-		deviceAffinity, err := GetAffinityForDevice(deviceScanCfg)
-		if err != nil {
-			log.Debugf("Error from GetAffinityForDevice: %v", err)
-			continue
+		log.Debugf("The target device is: %s", deviceScanCfg.targetDevice)
+		// Implements a workaround to handle the current psm2 provider behavior
```

Comment:
isn't this comment a duplication and repeated above?

Reply:
It depends on how you're using 'duplication'.  The code that does the device scanning and looks for a PSM2 device is similar between ScanFabric() and ValidateProviderConfig() but there are differences between them that made it compelling to make unique functions for these two similar services.